baseline/client/tests/n-back.test.js

```diff
@@ -0,0 +1,41 @@
+import { NBack } from "../n-back/n-back.js";
+import { pressKey } from "./utils.js"
+
+describe("n-back", () => {
+    it("results should have at least one result marked isRelevant", () => {
+        // check timeline nodes
+        const timeline = (new NBack()).getTimeline();
+        expect(timeline.some(trial => trial.data && trial.data.isRelevant)).toBe(true);
+        // check generated data
+        jest.useFakeTimers("legacy");
+        jsPsych.init({timeline: timeline});
+        pressKey(" ");  // skip introduction
+        pressKey(" ");  // skip overall instruction
+        pressKey(" "); jest.runAllTimers();  // skip 0-back instruction, cue, trial, and rest
+        pressKey(" "); jest.runAllTimers();  // skip 1-back instruction, cue, trial, and rest
+        pressKey(" "); jest.runAllTimers();  // skip 2-back instruction, cue, trial, and rest
+        pressKey(" ");  // skip completion
+        const progress = jsPsych.progress();
+        expect(progress.current_trial_global).toBe(progress.total_trials);
+        const relevant = jsPsych.data.get().filter({isRelevant: true}).values();
+        expect(relevant.length).toBeGreaterThan(0);
+    });
+    
+    it("trials are preceded by cues", () => {
+        const timeline = (new NBack()).getTimeline();
+        const trial_indices = Array.from(timeline.entries())
```

Comment:
Generally I'm a fan of the functional style, but this took me a while to understand. A couple of alternate suggestions:

More readable:
```
const n_back_trial_indices = [];
for (const [index, trial] of timeline.entries()) {
  if entry.type === 'n-back' {
     nBackIndices.push(index);
  }
}
```

More concise:
```
const reducer = (targetIndices, trial, index) => trial.type === 'n-back' ? targetIndices.concat([index]) : targetIndices;
const n_back_trial_indices = timeline.reduce(reducer, []);
```